project/core/management/commands/init_data.py

```diff
@@ -0,0 +1,128 @@
+import pprint
+import random
+from argparse import ArgumentTypeError
+from functools import partial
+
+from django.core.management.base import BaseCommand
+from factory import PostGenerationMethodCall, create_batch
+
+from users.factories import UserFactory
+from users.models import User
+from resources.factories import ResourceFactory
+from resources.models import Resource
+from tagging.factories import CustomTagFactory, TaggedItemsFactory
+from tagging.models import CustomTag, TaggedItems
+
+
+class Command(BaseCommand):
+
+    help = "Initialize the DB with some random fake data for testing and development"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--clear-db",
+            action="store_true",
+            dest="clear-db",
+            help="Clear existing data from the DB before creating test data",
+        )
+
+        def check_min_int(min, value):
+            ivalue = int(value)
+            if ivalue < min:
+                raise ArgumentTypeError(f"must be at least {min}")
+            return ivalue
+
+        parser.add_argument(
+            '--num-users',
+            type=partial(check_min_int, 1),
+            required=False,
+            default=10,
+            dest="num-users",
+            help="Number of `User` objects to create (default 10)"
+        )
+        parser.add_argument(
+            '--num-tags',
+            type=partial(check_min_int, 1),
+            required=False,
+            default=10,
+            dest="num-tags",
+            help="Number of `Tag` objects to create (default 10)"
+        )
+        parser.add_argument(
+            '--num-resources',
+            type=partial(check_min_int, 2),
+            required=False,
+            default=10,
+            dest="num-resources",
+            help="Number of `Resource` objects to create (default 10)"
+        )
+
+    def teardown(self):
+        TaggedItems.objects.all().delete()
+        CustomTag.objects.all().delete()
+        Resource.objects.all().delete()
+        User.objects.all().delete()
+
+    def create_users(self, num):
+        # Create random users.
+        # All users have the password 'codebuddies'.
```

Comment:
This is fancy :D And all those possible arguments! 😍 

One thing I noticed: I logged into adminer to check the new users that were created, and saw:

![image](https://user-images.githubusercontent.com/4512699/78855572-18d99c00-79d9-11ea-95ee-09414f254deb.png)

So I tried authenticating via one of those usernames and the password `codebuddies`, but was not able to do so. 

![image](https://user-images.githubusercontent.com/4512699/78855552-0c554380-79d9-11ea-9065-94ce6a08d14c.png)

Reply:
Thanks - well spotted. I've fixed this in dcca264 I'm not 100% sure why this fixes it. Maybe `create_batch()` doesn't fire the `post_generation` hook or something? Either way, its fixed..
I didn't notice this because I was originally creating users with a loop and then switched them to a `create_batch()` later :roll_eyes:

Reply:
(Sorry for slow reply -- thanks for fixing! :))